diff -urN gtkhtml-3.13.92/art/Makefile.am gtkhtml-3.13.92.new/art/Makefile.am
--- gtkhtml-3.13.92/art/Makefile.am	2007-02-09 20:54:57.000000000 +0100
+++ gtkhtml-3.13.92.new/art/Makefile.am	2007-03-04 15:53:20.000000000 +0100
@@ -4,7 +4,6 @@
 	info_bulb.png			\
 	empty_image.png			\
 	flowers.png			\
-	html-viewer.png			\
 	rule-blue-left.png		\
 	rule-blue-center.png		\
 	rule-blue-right.png		\
@@ -21,5 +20,9 @@
 	rect.png			\
 	texture.png
 
+pixmapsdir = $(datadir)/pixmaps
+pixmaps_DATA = html-viewer.png
+
 EXTRA_DIST = \
+	$(pixmaps_DATA)			\
 	$(icons_DATA)
