--- gtkhtml-1.0.4/src/htmlgdkpainter.c.fontsize	Tue Jul 23 21:20:53 2002
+++ gtkhtml-1.0.4/src/htmlgdkpainter.c	Tue Jul 23 21:22:21 2002
@@ -72,7 +72,7 @@
     gint n;
 
     /* Search paramether */
-    for (s=font_name, n=points ? 7 : 8; n; n--,s++)
+    for (s=font_name, n=points ? 8 : 7; n; n--,s++)
 	    s = strchr (s,'-');
 
     if (s && *s != 0) {
@@ -173,6 +176,21 @@
 	return rv;
 }
 
+static GdkFont *
+get_fallback_gdk_font (void)
+{
+	static GdkFont *fallback_font = NULL;
+
+	if (!fallback_font) {
+		GtkWidget *tmp_window = gtk_window_new (GTK_WINDOW_POPUP);
+		gtk_widget_ensure_style (tmp_window);
+		fallback_font = gdk_font_ref (tmp_window->style->font);
+		gtk_widget_destroy (tmp_window);
+	}
+
+	return fallback_font;
+}
+
 static gpointer
 alloc_e_font_try (gchar *face, gdouble size, gboolean points, GtkHTMLFontStyle style,
 		  gchar *medium, gchar *bold, gchar *roman, gchar *italic, gboolean known_size)
@@ -219,15 +237,7 @@
 
 		g_free (name);
 	} else {
-		GdkFont *fixed;
-
-		fixed = gdk_fontset_load ("fixed");
-		if (!fixed)
-			fixed = gdk_font_load ("fixed");
-		if (!fixed)
-			g_error (_("Can't load fixed font."));
-		font = e_font_from_gdk_font (fixed);
-		gdk_font_unref (fixed);
+		font = e_font_from_gdk_font (get_fallback_gdk_font ());
 	}
 
 
