diff -Nru gtkhtml-3.0.9.bak/src/Makefile.am gtkhtml-3.0.9/src/Makefile.am
--- gtkhtml-3.0.9.bak/src/Makefile.am	2003-09-04 00:42:09.000000000 +0200
+++ gtkhtml-3.0.9/src/Makefile.am	2003-09-25 10:47:00.432579464 +0200
@@ -203,7 +203,7 @@
 if HAVE_SOUP
 test_programs = testgtkhtml
 endif
-noinst_PROGRAMS = $(test_programs) gtest
+noinst_PROGRAMS = gtest
 
 testgtkhtml_SOURCES = 		\
 	testgtkhtml.c		\
