diff -urN gtkhtml-3.14.0/gtkhtml/Makefile.am gtkhtml-3.14.0.new/gtkhtml/Makefile.am
--- gtkhtml-3.14.0/gtkhtml/Makefile.am	2007-03-12 16:44:59.000000000 +0100
+++ gtkhtml-3.14.0.new/gtkhtml/Makefile.am	2007-03-13 17:45:16.000000000 +0100
@@ -207,7 +207,7 @@
 	$(NULL)
 
 libgtkhtml_3_14_la_LDFLAGS = -version-info $(GTKHTML_CURRENT):$(GTKHTML_REVISION):$(GTKHTML_AGE) -no-undefined
-libgtkhtml_3_14_la_LIBADD = $(GTKHTML_LIBS) ../a11y/libgtkhtml-a11y.la $(GAIL_LIBS) $(REGEX_LIBS)
+libgtkhtml_3_14_la_LIBADD = $(GTKHTML_LIBS) $(top_builddir)/a11y/libgtkhtml-a11y.la $(GAIL_LIBS) $(REGEX_LIBS)
 
 dist-hook:
 	mkdir $(distdir)/tests
