diff -urN aaa/gtkhtml-3.0.1/src/Makefile.am gtkhtml-3.0.1/src/Makefile.am
--- aaa/gtkhtml-3.0.1/src/Makefile.am	Thu Mar  6 05:50:52 2003
+++ gtkhtml-3.0.1/src/Makefile.am	Sat Mar  8 00:36:47 2003
@@ -193,19 +193,19 @@
 	cp -f $(srcdir)/tests/*.jpg $(distdir)/tests
 
 if HAVE_SOUP
-test_programs = testgtkhtml
+#test_programs = testgtkhtml
 endif
-noinst_PROGRAMS = $(test_programs) gtest
+noinst_PROGRAMS = gtest
 
-testgtkhtml_SOURCES = 		\
-	testgtkhtml.c		\
-	htmlurl.h		\
-	htmlurl.c
-testgtkhtml_LDFLAGS = -static
-testgtkhtml_LDADD =					\
-	libgtkhtml-@GTKHTML_RELEASE@.la			\
-	../a11y/libgtkhtml-a11y-@GTKHTML_RELEASE@.la 	\
-	$(SOUP_LIBS)
+#testgtkhtml_SOURCES = 		\
+#	testgtkhtml.c		\
+#	htmlurl.h		\
+#	htmlurl.c
+#testgtkhtml_LDFLAGS = -static
+#testgtkhtml_LDADD =					\
+#	libgtkhtml-@GTKHTML_RELEASE@.la			\
+#	../a11y/libgtkhtml-a11y-@GTKHTML_RELEASE@.la 	\
+#	$(SOUP_LIBS)
 
 gtest_SOURCES = \
 	test.c
