diff -Nru gtkhtml-0.12.0/art/Makefile.am gtkhtml-0.12.0.new/art/Makefile.am
--- gtkhtml-0.12.0/art/Makefile.am	Thu Aug 16 11:53:23 2001
+++ gtkhtml-0.12.0.new/art/Makefile.am	Sat Sep  8 12:56:44 2001
@@ -9,7 +9,6 @@
 	bulb.png			\
 	empty_image.png			\
 	flowers.png			\
-	html-viewer.png			\
 	insert-table-16.png		\
 	insert-table-24.png		\
 	insert-image-16.png		\
@@ -37,5 +36,9 @@
 	undo-16.png			\
 	undo-24.png
 
+pixmapsdir = $(datadir)/pixmaps
+pixmaps_DATA = html-viewer.png
+
 EXTRA_DIST = \
+	$(pixmaps_DATA) 		\
 	$(icons_DATA)
